plugins/org.yakindu.sct.doc.user/src/user-guide/embedded_systems_integration_guide.textile

```diff
@@ -0,0 +1,281 @@
+!(large)images/images_embedded_systems_integration_guide/platform_integration_example.jpeg(Example State Machine)!
+
+p=. Example State Machine
+
+The state machine contains three types of events: two __in events__ (inEvent1, inEvent2), two __out events__ (outEvent1, outEvent2) and one __timed event__ (after 2 s). Two states are connected with the inEvent1 and the after 2 s event. Each state raises one of __out event__. Additionally, a final state has been added, which also is part of the API. It only can be reached by raising the __in event__ inEvent2, if the state machine's __state__ StateA is active.
+
+These are the main parts of the state machine, which can be used and must be handled:
+* raise in events
+* raise timed events
+* handle out events
+* check if the state machine is final
+
+h2(#Bare-metal-cycle-based-integration). Bare-Metal Integration
+
+This section describes how to integrate a state machine with a cycle based or event driven execution semantic on any microcontroller, so on any embedded device. The integration can be structured in three parts, which results the main part of your program.
+* Initialization - called once
+* While loop - calling the state machine until it's final
+* Final - called if the state machine is final
+The execution of the program can be visualized as a state machine:
+
+!(large)images/images_embedded_systems_integration_guide/platform_integration_simple.png(High level description of bare metal cycle based state machine integration)!
+
+p=. High level description of bare metal cycle based state machine integration
+
+
+ According to the used State Machine, there are different parts, which must be take into account:
+* In Events 
+* Timed Events
+* Out Events
+
+h3(#main-function). Main Function
+
+The main function could be realized as following: First, the hardware will be initialized. Then the timers will take into account. After this the state machine will be initialized and entered. 
+At this point the state machine will be called as long as the state machine is not final. Note: If there is no final state defined the behavior is equals a while(true) loop. At first all in events will be raised. After this the timer will be updated, which internally raises time events. The behavior equals an in events. Then the runCycle will be called, which handles the logical execution of the state machine. This only needed, if the execution semantics is cycle based and must only be executed based on the elapsed time, e.g. 200 ms for a CycleBased(200) state machine. After proceeding, the out events will be handled. The last step, if wanted, is setting the microcontroller into a sleep mode.
+
+bc.. 
+void main() {
+	// Initialization
+	
+	hardwareInit();
+	
+	if(hasTimeEvents) {
+		timerInit();
+	}
+	
+	statemachine_init();
+	statemachine_enter();
+
+	// While loop
+	while(!statemachine_isFinal()) {
+		if(hasInEvents) {
+			handleInEvents();
+		}
+		
```

Comment:
We mix meta levels if we define it as a runtime expression. If we want to describe it in C-style then we shoudl use preprocessor directives.